quickwit/quickwit-control-plane/src/indexing_scheduler/scheduling/scheduling_logic_model.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::btree_map::Entry;
+use std::collections::BTreeMap;
+use std::num::NonZeroU32;
+
+pub type SourceOrd = u32;
+pub type NodeOrd = usize;
+pub type Load = u32;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Source {
+    pub source_id: SourceOrd,
```

Comment:
nitpick: replace `source_id` by `source_ord` to avoid confusion with the source ID.